webdriver/tests/document_handling/test_execute_script.py

```diff
@@ -0,0 +1,189 @@
+import pytest
+
+from tests.support.asserts import assert_same_element
+from tests.support.inline import inline
+from webdriver.client import element_key
+
+
+# Ported from Mozilla's marionette's test suite.
+
+# TODO:
+# * Return a window proxy
+# * Throw an exception during JS execution
+# * All error cases
+# * Probably more.
+
+def assert_elements_equal(session, expected, actual):
+    assert len(expected) == len(actual)
+    for e, a in zip(expected, actual):
+        e_id = e[element_key] if isinstance(e, dict) else e.json()[element_key]
+        a_id = a[element_key] if isinstance(a, dict) else a.json()[element_key]
+        assert e_id == a_id
+
+
+# All tests navigate to an HTML page to ensure that JS
+# can be executed. It's unknown whether the default start
+# page for all browsers allows JS execution.
+
+
+def test_return_number(session):
+    session.url = inline("""<title>JS testing</title>""")
+    assert 1 == session.execute_script("return 1")
+    assert 1.5 == session.execute_script("return 1.5")
+
+
+def test_return_boolean(session):
+    session.url = inline("""<title>JS testing</title>""")
+    assert True == session.execute_script("return true")
```

Comment:
Should be:

```py
response = execute_script("return true")
value = assert_success(response)
assert value is True
```

Reply:
how about:
`assert_success(response, True)`
?

Reply:
That will give a slightly worse error message if it fails, but would be acceptable.